### PR TITLE
Fix malformed metadata in extensions

### PR DESCRIPTION
**Pull request by majguo**
After upgrading to quarkus 3.0, some errors are observed when running dev mode for `integration-tests/azure-app-configuration` and `integration-tests/azure-storage-blob`.

<details>
  <summary>Error details:</summary>
  
  ```
  2023-06-09 16:29:04,545 ERROR [io.qua.dev.dep.DevUIProcessor] (build-8) Failed to process extension descriptor file:///mnt/c/Users/jiangma/Workspace/repos/quarkus-azure-services/internal/jackson-dataformat-xml/runtime/target/classes/META-INF/quarkus-extension.yaml: java.lang.RuntimeException: Failed to locate 'artifact' or 'group-id' and 'artifact-id' among metadata keys [name, description, metadata]
          at io.quarkus.devui.deployment.DevUIProcessor.getExtensionNamespace(DevUIProcessor.java:709)
          at io.quarkus.devui.deployment.DevUIProcessor.lambda$getAllExtensions$1(DevUIProcessor.java:425)
          at io.quarkus.runtime.util.ClassPathUtils.lambda$consumeAsPath$0(ClassPathUtils.java:121)
          at io.quarkus.runtime.util.ClassPathUtils.processAsPath(ClassPathUtils.java:161)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPath(ClassPathUtils.java:120)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPaths(ClassPathUtils.java:104)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPaths(ClassPathUtils.java:85)
          at io.quarkus.devui.deployment.DevUIProcessor.getAllExtensions(DevUIProcessor.java:409)
          at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
          at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
          at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
          at java.base/java.lang.reflect.Method.invoke(Method.java:566)
          at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
          at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
          at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
          at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
          at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
          at java.base/java.lang.Thread.run(Thread.java:829)
          at org.jboss.threads.JBossThread.run(JBossThread.java:501)
  
  2023-06-09 16:29:04,654 ERROR [io.qua.dev.dep.DevUIProcessor] (build-8) Failed to process extension descriptor file:///mnt/c/Users/jiangma/Workspace/repos/quarkus-azure-services/internal/core/runtime/target/classes/META-INF/quarkus-extension.yaml: java.lang.RuntimeException: Failed to locate 'artifact' or 'group-id' and 'artifact-id' among metadata keys [name, description, metadata]
          at io.quarkus.devui.deployment.DevUIProcessor.getExtensionNamespace(DevUIProcessor.java:709)
          at io.quarkus.devui.deployment.DevUIProcessor.lambda$getAllExtensions$1(DevUIProcessor.java:425)
          at io.quarkus.runtime.util.ClassPathUtils.lambda$consumeAsPath$0(ClassPathUtils.java:121)
          at io.quarkus.runtime.util.ClassPathUtils.processAsPath(ClassPathUtils.java:161)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPath(ClassPathUtils.java:120)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPaths(ClassPathUtils.java:104)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPaths(ClassPathUtils.java:85)
          at io.quarkus.devui.deployment.DevUIProcessor.getAllExtensions(DevUIProcessor.java:409)
          at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
          at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
          at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
          at java.base/java.lang.reflect.Method.invoke(Method.java:566)
          at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
          at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
          at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
          at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
          at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
          at java.base/java.lang.Thread.run(Thread.java:829)
          at org.jboss.threads.JBossThread.run(JBossThread.java:501)
  
  2023-06-09 16:29:04,677 ERROR [io.qua.dev.dep.DevUIProcessor] (build-8) Failed to process extension descriptor file:///mnt/c/Users/jiangma/Workspace/repos/quarkus-azure-services/internal/http-client-vertx/runtime/target/classes/META-INF/quarkus-extension.yaml: java.lang.RuntimeException: Failed to locate 'artifact' or 'group-id' and 'artifact-id' among metadata keys [name, description, metadata]
          at io.quarkus.devui.deployment.DevUIProcessor.getExtensionNamespace(DevUIProcessor.java:709)
          at io.quarkus.devui.deployment.DevUIProcessor.lambda$getAllExtensions$1(DevUIProcessor.java:425)
          at io.quarkus.runtime.util.ClassPathUtils.lambda$consumeAsPath$0(ClassPathUtils.java:121)
          at io.quarkus.runtime.util.ClassPathUtils.processAsPath(ClassPathUtils.java:161)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPath(ClassPathUtils.java:120)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPaths(ClassPathUtils.java:104)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPaths(ClassPathUtils.java:85)
          at io.quarkus.devui.deployment.DevUIProcessor.getAllExtensions(DevUIProcessor.java:409)
          at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
          at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
          at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
          at java.base/java.lang.reflect.Method.invoke(Method.java:566)
          at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
          at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
          at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
          at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
          at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
          at java.base/java.lang.Thread.run(Thread.java:829)
          at org.jboss.threads.JBossThread.run(JBossThread.java:501)
  
  2023-06-09 16:29:04,704 ERROR [io.qua.dev.dep.DevUIProcessor] (build-8) Failed to process extension descriptor file:///mnt/c/Users/jiangma/Workspace/repos/quarkus-azure-services/extensions/azure-storage-blob/runtime/target/classes/META-INF/quarkus-extension.yaml: java.lang.ClassCastException: class java.lang.String cannot be cast to class java.util.List (java.lang.String and java.util.List are in module java.base of loader 'bootstrap')
          at io.quarkus.devui.deployment.DevUIProcessor.lambda$getAllExtensions$1(DevUIProcessor.java:448)
          at io.quarkus.runtime.util.ClassPathUtils.lambda$consumeAsPath$0(ClassPathUtils.java:121)
          at io.quarkus.runtime.util.ClassPathUtils.processAsPath(ClassPathUtils.java:161)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPath(ClassPathUtils.java:120)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPaths(ClassPathUtils.java:104)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPaths(ClassPathUtils.java:85)
          at io.quarkus.devui.deployment.DevUIProcessor.getAllExtensions(DevUIProcessor.java:409)
          at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
          at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
          at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
          at java.base/java.lang.reflect.Method.invoke(Method.java:566)
          at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
          at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
          at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
          at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
          at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
          at java.base/java.lang.Thread.run(Thread.java:829)
          at org.jboss.threads.JBossThread.run(JBossThread.java:501)

  2023-06-12 14:54:33,156 ERROR [io.qua.dev.dep.DevUIProcessor] (build-28) Failed to process extension descriptor file:///mnt/c/Users/jiangma/Workspace/repos/quarkus-azure-services/extensions/azure-app-configuration/runtime/target/classes/META-INF/quarkus-extension.yaml: java.lang.ClassCastException: class java.lang.String cannot be cast to class java.util.List (java.lang.String and java.util.List are in module java.base of loader 'bootstrap')
          at io.quarkus.devui.deployment.DevUIProcessor.lambda$getAllExtensions$1(DevUIProcessor.java:438)
          at io.quarkus.runtime.util.ClassPathUtils.lambda$consumeAsPath$0(ClassPathUtils.java:121)
          at io.quarkus.runtime.util.ClassPathUtils.processAsPath(ClassPathUtils.java:161)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPath(ClassPathUtils.java:120)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPaths(ClassPathUtils.java:104)
          at io.quarkus.runtime.util.ClassPathUtils.consumeAsPaths(ClassPathUtils.java:85)
          at io.quarkus.devui.deployment.DevUIProcessor.getAllExtensions(DevUIProcessor.java:399)
          at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
          at java.base/java.lang.reflect.Method.invoke(Method.java:578)
          at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
          at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
          at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
          at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
          at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
          at java.base/java.lang.Thread.run(Thread.java:1589)
          at org.jboss.threads.JBossThread.run(JBossThread.java:501)
  ```
</details>

The PR is to resolve these errors by fixing malformed metadata in extensions, see references:
* [QUARKUS EXTENSION METADATA](https://quarkus.io/guides/extension-metadata)
* https://github.com/quarkusio/quarkus/issues/32414

Signed-off-by: Jianguo Ma <jiangma@microsoft.com>

### CODE REVIEW

File: extensions/azure-storage-blob/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 1]
```diff
@@ -1,14 +1,16 @@
+---
 name: "Quarkus Support Azure Storage Blob"
-description: "Quarkus Support for Azure Storage Blob"
```
Comment (backwind1233) [inline extensions/azure-storage-blob/runtime/src/main/resources/META-INF/quarkus-extension.yaml:2]: Can we keep this "description" part?

Comment (majguo) [reply extensions/azure-storage-blob/runtime/src/main/resources/META-INF/quarkus-extension.yaml:2]: Good catch. From [QUARKUS EXTENSION METADATA](https://quarkus.io/guides/extension-metadata), the `description` can be provided in the template file. However, in our case, the content of the `description` is exactly/almost the same as `name`, that's why I get them removed.

Comment (backwind1233) [reply extensions/azure-storage-blob/runtime/src/main/resources/META-INF/quarkus-extension.yaml:2]: I think Spring Cloud Azure has the same case as quarkus, you can check this [link](https://github.com/Azure/azure-sdk-for-java/blob/5032c95603837463c1647b1af6df2e048f129870/sdk/spring/spring-cloud-azure-autoconfigure/pom.xml#L17).

image

but I'd rather to keep them both.

Comment (majguo) [reply extensions/azure-storage-blob/runtime/src/main/resources/META-INF/quarkus-extension.yaml:2]: Fair enough. Fixed with commit [add back description](https://github.com/quarkiverse/quarkus-azure-services/pull/116/commits/093a08ec82124847fda541cca242ff0a15209e16).

**Review by edburns — APPROVED** [review]

**Review by backwind1233 — APPROVED** [review]
LGTM

**Comment by majguo** [comment]
@all-contributors
please add @edburns for review. 
please add @backwind1233 for review.

**Comment by majguo** [comment]
@all-contributors please add @backwind1233 for review.

**Pull request merged**